calculation/sha_calc/sha_calc/directivity/bea20/bea20.py

```diff
@@ -0,0 +1,226 @@
+import math
+from typing import List
+
+import numpy as np
+import numpy.matlib
+
+
+def bea20(
+    M: float,
+    U: np.ndarray,
+    T: np.ndarray,
+    Smax: List,
+    D: float,
+    Tbot: float,
+    Dbot: float,
+    Rake: float,
+    Dip: float,
+    Period: float,
+    type: int = 0,
+):
+    """
+    Calculates the directivity effects at given locations based on fault information.
+    Algorithm has been taken from the Bayless 2020 model found in the paper.
+    "A Rupture Directivity Adjustment Model Applicable to the NGA-West2 Ground Motion Models and Complex Fault Geometries
+    (Bayless 2020)".
+    This functionality was compared to the matlab code taken from the paper.
+
+    Parameters
+    ----------
+    M: float
+        Moment magnitude, 5<=M<=8
+    U: np.ndarray
+        The GC2 coordinates in km. Equivalent to ry.
+        Must be nX1 number where n is the number of locations at which the model provides a prediction.
+    T: np.ndarray
+        The GC2 coordinates in km. Equivalent to rx.
+        Must be nX1 number where n is the number of locations at which the model provides a prediction.
+    Smax: List
+        List of 2x1 number where first is the lowest max value and the second is the largest max value for s.
+        Computed from GC2 using nominal strike of the fault.
+    D: float
+        The effective rupture travel width, measured from the hypocenter to the shallowest depth of the rupture plane.
+    Tbot: float
+        The length of the bottom of the rupture plane (projected to the surface) in km.
+    Dbot: float
+        The vertical depth of the bottom of the rupture plane from the ground surface, including Ztor, in km.
+    Rake: float
+        The Rake of the fault
+    Dip: float
+        The Dip of the fault at the hypocenter
+    Period: float
+        Used for specifying the period for a pSA IM
+    type: float, optional
+        0 for type base don strike
+        1 for a strike slip
+        2 for a oblique, reverse or normal
+    """
+
+    # If not specified, determine rupture category from Rake angle
+    if type == 0:
+        if (-30 <= Rake <= 30) or (-180 <= Rake <= -150) or (150 <= Rake <= 180):
+            type = 1
+        else:
+            type = 2
+
+    # Convert U to S
+    # Limit S to the Smax values for positive and negative numbers
+    Smax1, Smax2 = Smax
+    U, S = np.asarray(U).copy(), np.asarray(U).copy()
+    S[U < Smax1] = Smax1
+    S[U > Smax2] = Smax2
+
+    # Convert U to Ry
+    # Reduce each positive and negative side by absolute value of Smax and remove negative numbers inbetween
+    upos, uneg = U >= 0, U < 0
+    Ry = np.zeros(U.size)
```

Comment:
I think this block can be made a bit shorter & clearer (imo) using `np.where`
```
Ry = np.where(U >= 0, U[U >= 0] - Smax2, np.abs(U[U < 0]) - abs(Smax1))
Ry[Ry < 0] = 0
```

One other comment, you are using `abs` here on an numpy array, I'm pretty sure this works as numpy overwrites the built-in `abs` function. However I think in general its clearer to use `np.abs` when working on arrays